src/test/java/htsjdk/samtools/CRAMAllEncodingStrategiesTest.java

```diff
@@ -21,32 +21,72 @@ public class CRAMAllEncodingStrategiesTest extends HtsjdkTest {
 
-    @DataProvider(name="roundTripTestFiles")
-    public Object[][] roundTripTestFiles() {
+    @DataProvider(name="defaultStrategyRoundTripTestFiles")
+    public Object[][] defaultStrategyRoundTripTestFiles() {
         return new Object[][] {
                 { new File(TEST_DATA_DIR, "NA12878.20.21.1-100.100-SeqsPerSlice.500-unMapped.cram"),
-                        new File(TEST_DATA_DIR, "human_g1k_v37.20.21.1-100.fasta") },
+                        new File(TEST_DATA_DIR, "human_g1k_v37.20.21.1-100.fasta"),
+                        false, false },
+                { new File(TEST_DATA_DIR, "CEUTrio.HiSeq.WGS.b37.NA12878.20.21.10m-10m100.cram"),
+                        new File("src/test/resources/htsjdk/samtools/reference/human_g1k_v37.20.21.fasta.gz"),
+                        false, false },
+                { new File(TEST_DATA_DIR, "NA12878.unmapped.cram"),
+                        new File(TEST_DATA_DIR, "human_g1k_v37.20.21.1-100.fasta"),
+                        false, false },
+                // generated with samtools 1.19 from the gatk bam file CEUTrio.HiSeq.WGS.b37.NA12878.20.21.bam
+                { new File(TEST_DATA_DIR, "CEUTrio.HiSeq.WGS.b37.NA12878.20.21.v3.0.samtools.cram"),
+                        new File("src/test/resources/htsjdk/samtools/reference/human_g1k_v37.20.21.fasta.gz"),
+                        true, false },
+                // these tests use lenient equality to only validate read names, bases and qual scores
+                { new File(TEST_DATA_DIR, "mitoAlignmentStartTestGATKGen.cram"),
+                        new File(TEST_DATA_DIR, "mitoAlignmentStartTest.fa"), true, false },
+                { new File(TEST_DATA_DIR, "mitoAlignmentStartTest.cram"),
+                        new File(TEST_DATA_DIR, "mitoAlignmentStartTest.fa"), true, false },
+                // files created by rewriting the htsjdk test file src/test/resources/htsjdk/samtools/cram/mitoAlignmentStartTest.cram
+                // using code that replicates the first read (which is aligned to position 1 of the mito contig) either
+                // 10,000 or 20,000 times, to create a file with 2 or 3 containers, respectively, that have reads aligned to
+                // position 1 of the contig
+                { new File(TEST_DATA_DIR, "mitoAlignmentStartTest_2_containers_aligned_to_pos_1.cram"),
+                        new File(TEST_DATA_DIR, "mitoAlignmentStartTest.fa"), true, false },
+                { new File(TEST_DATA_DIR, "mitoAlignmentStartTest_3_containers_aligned_to_pos_1.cram"),
+                        new File(TEST_DATA_DIR, "mitoAlignmentStartTest.fa"), true, false }
         };
     }
 
-    @Test(dataProvider = "roundTripTestFiles")
-    public final void testRoundTripDefaultEncodingStrategy(final File sourceFile, final File referenceFile) throws IOException {
+    @Test(dataProvider = "defaultStrategyRoundTripTestFiles")
+    public final void testRoundTripDefaultEncodingStrategy(
```

Comment:
Add a comment before this test method describing the purpose of the test (since we're not just testing "encoding strategies" here....)

Reply:
Its testing the default encoding strategy. Comment added.